File: frontend/replace-classes.js
const fs = require('fs');
const path = require('path');

const dirs = [
  path.join(__dirname, 'src', 'pages'),
  path.join(__dirname, 'src', 'components'),
  path.join(__dirname, 'src', 'utils'),
];

// Replacement map: [pattern, replacement]
// Order matters - more specific patterns first
const replacements = [
  // Opacity variants (before base replacements)
  [/bg-stone-50\/50/g, 'bg-warm/10'],
  [/bg-stone-100\/50/g, 'bg-warm/15'],

  // Emerald replacements (from user mapping)
  [/bg-emerald-700/g, 'bg-primary'],
  [/bg-emerald-600/g, 'bg-primary/90'],
  [/bg-emerald-50/g, 'bg-primary/5'],
  [/text-emerald-700/g, 'text-primary'],
  [/text-emerald-600/g, 'text-primary'],
  [/text-emerald-500/g, 'text-primary/70'],
  [/text-emerald-400/g, 'text-primary/60'],
  [/hover:bg-emerald-50/g, 'hover:bg-primary/5'],
  [/hover:text-emerald-600/g, 'hover:text-primary'],
  [/border-emerald-200/g, 'border-primary/20'],
  [/border-emerald-300/g, 'border-primary/30'],

  // Stone replacements (from user mapping)
  [/bg-stone-100/g, 'bg-warm/30'],
  [/bg-stone-50/g, 'bg-warm/30'],
  [/text-stone-700/g, 'text-text-primary'],
  [/text-stone-600/g, 'text-text-primary'],
  [/text-stone-500/g, 'text-text-muted'],
  [/text-stone-400/g, 'text-text-muted'],
  [/text-stone-300/g, 'text-text-muted/50'],
  [/border-stone-200/g, 'border-border'],
  [/border-stone-100/g, 'border-border/50'],

  // Additional patterns per user instructions
  [/text-stone-800/g, 'text-text-primary'],
  [/hover:bg-stone-100/g, 'hover:bg-warm/30'],
  [/hover:bg-stone-50/g, 'hover:bg-warm/30'],
  [/hover:text-stone-700/g, 'hover:text-text-primary'],
  [/hover:text-stone-600/g, 'hover:text-text-primary'],
  [/hover:border-stone-200/g, 'hover:border-border'],
  [/border-emerald-500/g, 'border-primary'],
  [/focus:border-emerald-400/g, 'focus:border-primary'],
  [/accent-emerald-600/g, 'accent-primary'],
  [/border-emerald-100/g, 'border-primary/10'],
  [/text-emerald-800/g, 'text-primary'],
  [/hover:bg-emerald-600/g, 'hover:bg-primary/90'],
  [/hover:bg-emerald-700/g, 'hover:bg-primary'],
  [/hover:border-emerald-200/g, 'hover:border-primary/20'],

  // Missing stone patterns
  [/border-stone-50/g, 'border-border/50'],
  [/hover:border-stone-300/g, 'hover:border-border'],

  // Layout replacements
  [/rounded-2xl/g, 'rounded-card'],
  [/shadow-sm/g, 'shadow-card'],
  [/shadow-md/g, 'shadow-card'],
  [/rounded-xl/g, 'rounded-card'],
  [/rounded-lg/g, 'rounded-btn'],
];

function processFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf-8');
  let changed = false;
  for (const [pattern, replacement] of replacements) {
    const newContent = content.replace(pattern, replacement);
    if (newContent !== content) {
      changed = true;
      content = newContent;
    }
  }
  if (changed) {
    fs.writeFileSync(filePath, content, 'utf-8');
    console.log(`Updated: ${filePath}`);
  } else {
    console.log(`No changes: ${filePath}`);
  }
}

// Get all .tsx files from the directories
for (const dir of dirs) {
  if (fs.existsSync(dir)) {
    const files = fs.readdirSync(dir).filter(f => f.endsWith('.tsx'));
    for (const file of files) {
      processFile(path.join(dir, file));
    }
  }
}

console.log('Done!');

File: frontend/replace-classes.py
#!/usr/bin/env python3
"""批量替换 Tailwind CSS 类名为新设计系统令牌"""
import os
import re

BASE_DIR = os.path.dirname(os.path.abspath(__file__))
DIRS = [
    os.path.join(BASE_DIR, 'src', 'pages'),
    os.path.join(BASE_DIR, 'src', 'components'),
]

# 替换规则： (模式字符串, 替换字符串)
# 更具体的规则在前
REPLACEMENTS = [
    # Opacity variants (before base replacements)
    (r'bg-stone-50/50', 'bg-warm/10'),
    (r'bg-stone-100/50', 'bg-warm/15'),

    # Emerald replacements
    (r'bg-emerald-700', 'bg-primary'),
    (r'bg-emerald-600', 'bg-primary/90'),
    (r'bg-emerald-50', 'bg-primary/5'),
    (r'text-emerald-700', 'text-primary'),
    (r'text-emerald-600', 'text-primary'),
    (r'text-emerald-500', 'text-primary/70'),
    (r'text-emerald-400', 'text-primary/60'),
    (r'hover:bg-emerald-50', 'hover:bg-primary/5'),
    (r'hover:text-emerald-600', 'hover:text-primary'),
    (r'border-emerald-200', 'border-primary/20'),
    (r'border-emerald-300', 'border-primary/30'),

    # Stone replacements
    (r'bg-stone-100', 'bg-warm/30'),
    (r'bg-stone-50', 'bg-warm/30'),
    (r'text-stone-700', 'text-text-primary'),
    (r'text-stone-600', 'text-text-primary'),
    (r'text-stone-500', 'text-text-muted'),
    (r'text-stone-400', 'text-text-muted'),
    (r'text-stone-300', 'text-text-muted/50'),
    (r'border-stone-200', 'border-border'),
    (r'border-stone-100', 'border-border/50'),

    # Additional patterns
    (r'text-stone-800', 'text-text-primary'),
    (r'hover:bg-stone-100', 'hover:bg-warm/30'),
    (r'hover:bg-stone-50', 'hover:bg-warm/30'),
    (r'hover:text-stone-700', 'hover:text-text-primary'),
    (r'hover:text-stone-600', 'hover:text-text-primary'),
    (r'hover:border-stone-200', 'hover:border-border'),
    (r'border-emerald-500', 'border-primary'),
    (r'focus:border-emerald-400', 'focus:border-primary'),
    (r'accent-emerald-600', 'accent-primary'),
    (r'border-emerald-100', 'border-primary/10'),

    # Additional emerald hover
    (r'hover:border-emerald-200', 'hover:border-primary/20'),
    (r'hover:border-emerald-300', 'hover:border-primary/30'),
    (r'hover:border-emerald-400', 'hover:border-primary'),
    (r'hover:bg-emerald-600', 'hover:bg-primary/90'),
    (r'hover:bg-emerald-700', 'hover:bg-primary'),
    (r'hover:bg-emerald-500', 'hover:bg-primary/70'),

    # Additional stone hover
    (r'hover:border-stone-300', 'hover:border-border'),
    (r'hover:border-stone-400', 'hover:border-border'),
    (r'hover:bg-stone-200', 'hover:bg-warm/40'),

    # Layout replacements
    (r'shadow-sm', 'shadow-card'),
    (r'shadow-md', 'shadow-card'),
    (r'rounded-xl', 'rounded-card'),
    (r'rounded-lg', 'rounded-btn'),
    (r'rounded-2xl', 'rounded-card'),
]

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    changed = False
    for pattern, replacement in REPLACEMENTS:
        new_content = re.sub(pattern, replacement, content)
        if new_content != content:
            changed = True
            content = new_content

    if changed:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f'Updated: {filepath}')
    else:
        print(f'No changes: {filepath}')


for d in DIRS:
    if os.path.exists(d):
        for fname in os.listdir(d):
            if fname.endswith('.tsx'):
                process_file(os.path.join(d, fname))

print('Done!')
